internal/controllers/clusterctl/config.yaml

```diff
@@ -21,6 +21,9 @@ data:
     - name:         "azure"
       url:          "https://github.com/rancher-sandbox/cluster-api-provider-azure/releases/v1.17.0/infrastructure-components.yaml"
       type:         "InfrastructureProvider"
+    - name:         "gcp"
+      url:          "https://github.com/kubernetes-sigs/cluster-api-provider-gcp/releases/v1.8.0/infrastructure-components.yaml"
```

Comment:
Should this repo be included in forks build?

Reply:
Yes, correct. I have this in my TODO list but I wanted to have it all working first and then I'll create a fork and build process like we do with other providers.

Reply:
Added a new issue to address this as a follow-up https://github.com/rancher/highlander/issues/109

Reply:
Should we wait until images from the fork are built (looks like they soon be available?) first and update the ref here, before taking this PR in?

Reply:
could it be related to some kustomization where it misses image patching, i.e https://github.com/kubernetes-sigs/cluster-api/blob/main/Makefile#L1319-L1322?

Reply:
@salasberryfin eventually I believe the source of truth for image as I checked is:

` image: registry.k8s.io/cluster-api-gcp/cluster-api-gcp-controller:v1.8.0`

coming from https://github.com/rancher-sandbox/cluster-api-provider-gcp/blob/main/Makefile#L120-L121 and that ends up eventually without modification in the release artifact files

Reply:
Alright, I found the root cause of it:

For CAPI manifests, it looks fine:

`image: registry.rancher.com/rancher/cluster-api-controller:v1.8.4`

because Makefile definition of it uses `?` mark in var: https://github.com/rancher-sandbox/cluster-api/blob/main/Makefile#L203C1-L203C45

resulting in image patching correctly, since we explicitly pass `PROD_REGISTRY` in release workflow: https://github.com/rancher/clusterapi-forks/blob/main/.github/workflows/publish_release/action.yaml#L51C9-L51C22


However, both [CAPZ](https://github.com/rancher-sandbox/cluster-api-provider-azure/blob/main/Makefile#L162)/[CAPG](https://github.com/rancher-sandbox/cluster-api-provider-gcp/blob/main/Makefile#L120) omits `?`, hence image is not overwritten as expected.

Reply:
We will tackle possibility of image overriding in upstream CAPZ/CAPG repos.